jbmc/src/jbmc/jbmc_parse_options.cpp

```diff
@@ -250,11 +250,13 @@ void jbmc_parse_optionst::get_command_line_options(optionst &options)
     options.set_option("refine-strings", false);
 
   if(cmdline.isset("string-printable"))
-    options.set_option("string-printable", true);
-
-  if(cmdline.isset("no-refine-strings") && cmdline.isset("string-printable"))
   {
-    warning() << "--string-printable ignored due to --no-refine-strings" << eom;
+    if(cmdline.isset("no-refine-strings"))
```

Comment:
Good point. I've split this into two commits, since I think the two refactorings related to initialize_nondet_string_fields are too closely related to be done separately: one changes the caller to handle the informal preconditions, and one changes the callee to turn informal preconditions to proper preconditions.